src/java.base/share/classes/java/net/DatagramSocket.java

```diff
@@ -422,6 +422,10 @@ public void bind(SocketAddress addr) throws SocketException {
      * call to send or receive may throw a PortUnreachableException. Note,
      * there is no guarantee that the exception will be thrown.
      *
+     * <p> If this socket is already connected, then this method will attempt to
+     * connect to the passed address and if that attempt completes exceptionally,
+     * then this socket will continue to be connected to the previously connected address.
```

Comment:
I don't know if there are other implementations DatagramSocketImpl in the eco system but the text could potentially invalidate them.  I think it would be better to say that if connect fails then the state of the DatagramSocket is unknown, it may or may not be connected to the address that it was previously connected to.

Reply:
Hello Alan,

> I don't know if there are other implementations DatagramSocketImpl in the eco system but the text could potentially invalidate them.

I hadn't taken into account other implementations. I've now updated the PR to use the text that you suggested.